sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -671,7 +675,7 @@ def state(self):
     return (api_jobstate_map[self._job.currentState] if self._job.currentState
             else PipelineState.UNKNOWN)
 
-  def _is_in_terminal_state(self):
+  def is_in_terminal_state(self):
```

Comment:
Let's keep the underscore--if we remove it here, it becomes part of the public API for DataflowRunner, and we can't change it back without a reasonable deprecation period.  You can still call it from the DataflowMetrics code since it is an implementation detail of the DataflowRunner and suppress any lint warnings.